crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -227,6 +227,22 @@ impl UpdateTask {
 }
 
 impl CachingNativePriceEstimator {
+    /// Convert from normalized price to floating point price
+    fn from_normalized_price(price: U256) -> Option<f64> {
```

Comment:
I think this conversion should happen in the `autopilot/database` module. With DDD we want to have the parts outside the `domain` to cater to its needs instead of having the `domain` take whatever the DB returns and make it work.